pynicotine/plugins/core_commands/__init__.py

```diff
@@ -26,39 +26,637 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self.commands = {
+            "help": {
+                "aliases": ["?"],
+                "callback": self.help_command,
+                "description": "List commands",
+                "usage": ["[query]"]
+            },
+            "rescan": {
+                "callback": self.rescan_command,
+                "description": _("Rescan shares"),
+                "group": _("Shares"),
+                "usage": ["[-force]", ""]
+            },
+            "hello": {
+                "aliases": ["echo", "greet"],
+                "callback": self.hello_command,
+                "description": "Print something",
+                "group": _("Message"),
+                "usage": ["[something..]"]
+            },
+            "away": {
+                "aliases": ["a"],
+                "callback": self.away_command,
+                "description": _("Toggle away status")
+            },
+            "plugin": {
+                "callback": self.plugin_handler_command,
+                "description": _("Load or unload a plugin"),
+                "usage": ["<toggle|enable|disable>", "<plugin_name>"]
+            },
+            "quit": {
+                "aliases": ["q", "exit"],
+                "callback": self.quit_command,
+                "description": _("Quit Nicotine+"),
+                "usage": ["[-force]", ""]  # "" disallow extra args
+            },
+            "clear": {
+                "aliases": ["cl"],
+                "callback": self.clear_command,
+                "description": _("Clear chat window"),
+                "disable": ["cli"],
+                "group": _("Chat"),
+                "usage": [""]  # "" disallow any args
+            },
+            "ctcp": {
+                "callback": self.ctcp_command,
+                "description": _("Send client-to-client protocol query"),
+                "disable": ["cli"],
+                "group": _("Chat"),
+                "usage": ["<version|time|ping>", "[user]"]
```

Comment:
I don't like the idea of sending times. I'm concerned it would expose your approximate country when behind a VPN.

Reply:
My intention is to move CTCP functions into a core_commands module (ctcp_commands.py), and expose it's options in a Core Commands plugin settings settings dialog. The replies can be handled in there with an incomming_chat_event() too.

Reply:
If it were a seperate function, then it would be disabled by default and users would be unlikely to enable it. We dont want the core_commands to be disableable.